mesh_tensorflow/signal_ops.py

```diff
@@ -101,46 +125,22 @@ class iFFT3DOperation(mtf.Operation):
   Returns:
     A Tensor of shape `input.shape[:-3] + dims`.
   """
-  def __init__(self, input,  dims, name=None):
-    super(iFFT3DOperation, self).__init__([input], name=name or "iFFT3D")
-    self._dims = dims
-    self._output_shape = mtf.Shape(input.shape[:-3]+dims)
-    self._outputs = [mtf.Tensor(self, mtf.Shape(self._output_shape), input.dtype)]
+  def __init__(self, inputs,  dims, name=None):
+    super(iFFT3DOperation, self).__init__(inputs, dims, inverse=True, name=name)
+
+  def _transpose(self, mesh_impl, split_axes, slices, naxes):
```

Comment:
same comment, if we move transpose in the base class, no need for this derived class